android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -209,21 +184,63 @@ public void run() {
           case FADE:
             transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
             break;
+          case SLIDE:
+            customAnimation = true;
+            getOrCreateTransaction().setCustomAnimations(R.anim.open_slide_in, R.anim.open_slide_out);
+            break;
+        }
+
+        if (!customAnimation) {
+          getOrCreateTransaction().setTransition(transition);
         }
-        getOrCreateTransaction().setTransition(transition);
       }
     } else if (mTopScreen != null && !mTopScreen.equals(newTop)) {
       // otherwise if we are performing top screen change we do "back animation"
       int transition = FragmentTransaction.TRANSIT_FRAGMENT_CLOSE;
+
       switch (mTopScreen.getScreen().getStackAnimation()) {
         case NONE:
           transition = FragmentTransaction.TRANSIT_NONE;
           break;
         case FADE:
           transition = FragmentTransaction.TRANSIT_FRAGMENT_FADE;
           break;
+        case SLIDE:
+          customAnimation = true;
+          getOrCreateTransaction().setCustomAnimations(R.anim.close_slide_in, R.anim.close_slide_out);
+          break;
+      }
+
+      if (!customAnimation) {
+        getOrCreateTransaction().setTransition(transition);
+      }
+    }
+
+    // remove all screens previously on stack
```

Comment:
This portion is moved from the top because the animations need to be set first before any screens are added/removed